### PR TITLE
feat(analytics): add daily activation rollups

### LINKED ISSUE #137: feat(analytics): add daily rollups and retention funnels
## Background

Raw product events are useful only if they become stable adoption and retention metrics. Rollups should show whether users actually reach a valuable first action.

## Goal

Aggregate product events into daily rollups and activation funnels.

## Current Behavior

There is no daily product usage rollup or activation funnel model.

## Desired Behavior

Operators can see activation and retention across MCP, GitHub App, API, and control-panel surfaces.

## Implementation Requirements

- Create daily rollups for users, repos, commands, tools, route classes, and outcomes.
- Track login to doctor pass to first useful action.
- Track GitHub install to first command to first useful maintainer action.
- Make aggregation idempotent and resilient to late events.
- Expose stale or incomplete rollup status.

## Public/Private Output Boundaries

Public GitHub output must never include wallets, hotkeys, payout/reward estimates, raw trust scores, public score estimates, private reviewability, private scoreability context, or farming language. Private API/MCP/control-panel surfaces may show authenticated scoreability, blockers, projections, and evidence, but must not claim guaranteed payout outcomes.

## Acceptance Criteria

- Tracks activation: login -> doctor pass -> first useful action.
- Tracks GitHub activation: install -> first command -> first useful maintainer action.
- Worker-safe aggregation.
- Tests cover idempotent rollups and late events.

## Testing Requirements

- `npm run test:ci` must pass.
- Global coverage must remain at or above 97% for lines, statements, functions, and branches.
- Aim for 98%+ branch coverage locally.
- Add tests for every new branch, fallback path, sanitizer rule, and regression.
- Add invariant/property-style tests when behavior depends on sorting, gating, scoring, source-upload safety, public/private boundaries, or telemetry privacy.
- MCP/local tooling must prove source contents are not uploaded when local metadata is involved.
- Public GitHub output must be tested against forbidden language: wallet, hotkey, raw trust score, payout, reward estimate, farming, private reviewability, and public score estimate.

### PR DESCRIPTION
**Pull request by oktofeesh1**
## Summary
- Adds daily product usage rollups with activation funnels built from the sanitized usage-event spine.
- Wires scheduled/internal rollup jobs plus private analytics API and operator-dashboard exposure.
- Adds an analytics UI table and regression coverage for idempotency, stale/incomplete days, route classes, guards, and webhook telemetry.

## What changed
- Added `product_usage_daily_rollups` storage and typed rollup records/status.
- Computes activation funnels for login, MCP doctor/useful actions, GitHub installation, command, and maintainer-useful activity.
- Marks current days partial, capped days incomplete, and late raw events stale until rerun.
- Adds `/v1/app/analytics/daily-rollups` and queued/immediate internal rollup job endpoints.
- Records bounded GitHub installation telemetry through the existing sanitized usage-event path.
- Shows daily activation rollups in the analytics UI.

## Why
Closes #137.

This gives downstream adoption/reporting work a stable aggregate layer instead of forcing dashboards and reports to re-scan raw product events.

## Validation
- `npm run test -- test/unit/product-usage.test.ts test/unit/queue.test.ts test/integration/api.test.ts test/integration/routes-errors.test.ts test/unit/index.test.ts`
- `npm run typecheck`
- `npm run test:coverage`
- `npm run ui:lint`
- `npm run test:ci`
- Codex Security diff scan: no findings

## Notes
- Draft because this is stacked on #182. Review after #182, or expect this diff to include the usage-event spine until #182 lands.
- Rollups aggregate sanitized product usage events and store aggregate counts/dimensions, not raw actors, tokens, source, or local paths.

### CODE REVIEW

**Review by JSONbored — APPROVED** [review]
@oktofeesh1 This is ready from my side. The daily activation rollup work is now green and merge-clean against current `main`.

A few notes:
- The migration, repository methods, queue scheduling, API route coverage, and UI analytics wiring are included.
- The rollup behavior has direct unit and integration coverage.
- This should land before dependent analytics/reporting branches that build on daily usage aggregates.

File: src/api/routes.ts [hunk 1]
```diff
@@ -713,12 +790,21 @@ export function createApp() {
         { label: "Installed coverage", value: installedRepos, spark: sparklineFromCounts(installedRepos, Math.max(repositories.length, 1)) },
       ],
       weeklyReport: buildOperatorWeeklyReport({ repositories, installations, health, registry, scoring, upstreamDrift }),
+      usageSummary,
+      usageRollups,
+      usageRollupStatus,
       registry,
       scoringModel: scoring,
       upstreamDrift,
     });
```
[the remark this answers is not in the document]
Comment (oktofeesh1) [reply src/api/routes.ts:799]: This is addressed in the current PR head. `/v1/app/analytics/daily-rollups` now calls `requireAppRole(c, ["operator"])` before reading rollups, and integration coverage asserts non-operator sessions receive `403` for this endpoint.

Validation: `npm run test -- test/integration/api.test.ts test/unit/product-usage.test.ts test/unit/queue.test.ts test/unit/openapi.test.ts` (54 tests passing).
